### PR TITLE
Encoding in AbstractHttpRequestExecutingMessageHandler.java is now done when flag is set

### PR DESCRIPTION
**Pull request by ezylot**
Before we did not actually encode the URI. You can see this in the screenshot below, where the String "häuser" is not actually encoded to the correct percentage encoding.

withoutEncode

When we first encode it and then call the .toUri() we can see that the "häuser" is now propoerly encoded to "h%C3%A4user"

withEncode

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ezylot Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2929)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2929) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ezylot Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2929)!

**Review by artembilan — CHANGES_REQUESTED** [review]
Please, add your name to the `@author` list and modify, please, `HttpRequestExecutingMessageHandlerTests.testUriExpression()` to reflect your change. 
This way we definitely going to have a test coverage.

Thank you!

**Comment by ezylot** [comment]
I added myself as an author and added 2 tests that check this specific problem

File: spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java [hunk 1]
```diff
@@ -27,11 +27,7 @@
 import java.io.Serializable;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (artembilan) [inline spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java]: No asterisk imports at all, please.

**Review by artembilan — CHANGES_REQUESTED** [review]
Please, fix Checkstyle violation

File: spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java (hunk 1 above)
Comment (ezylot) [reply spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java]: Oh boy, did not check that. Fixed.

**Review by artembilan — APPROVED** [review]
LGTM

Will  merge after green Travis report.

**Comment by artembilan** [comment]
There are still a couple Checkstyle violations:
```
[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java:770:45: '}' at column 31 should be alone on a line. [RightCurly]
[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-http/src/test/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandlerTests.java:792:45: '}' at column 31 should be alone on a line. [RightCurly]
```

**Comment by ezylot** [comment]
Damnit, give me a second ^^

**Comment by ezylot** [comment]
should be good to go, i think ;)

**Comment by artembilan** [comment]
Sorry, have missed your fresh commit somehow...

Now it is merged and I'm going to cherry-pick it to `5.1.x`.

Thank you for contribution; looking forward for more!

**Pull request merged**